### PR TITLE
RTD: Add configuration file for Read the Docs, which is mandatory now

### PR DESCRIPTION
**Pull request by amotl**
## About

Read the Docs now mandates a configuration file, `.readthedocs.yml`.

## Problem

Otherwise, RTD builds will fail.

image

-- https://readthedocs.org/projects/crate/builds/22054840/

## Details

This is applicable to both the `4.8` and `3.3` branches.

See commits as several fixes are required to make the build work for `4.8`.

### CODE REVIEW

File: .readthedocs.yml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+---
+version: 2
+
+build:
+  os: "ubuntu-22.04"
+  tools:
+    python: "3.11"
```
Comment (matriv) [inline .readthedocs.yml:7]: Just double checking, is 3.11 ok for those old versions?

**Review by matriv — APPROVED** [review]
Thx! Left a question.

File: .readthedocs.yml (hunk 1 above)
Comment (seut) [reply .readthedocs.yml:7]: Good point, I think we should at least verify that our oldest branch at RTD, 3.3, works with 3.11 (I think 3.6 was used back then).

Comment (amotl) [reply .readthedocs.yml:7]: It looks like it works here. The backport will tell us if it works on 3.3 as well.

image

Comment (amotl) [reply .readthedocs.yml:7]: Because it will need manual pushing to get this past failing CI steps, I need your support to bring it in, @seut.

Comment (seut) [reply .readthedocs.yml:7]: Looks like we need to backport https://github.com/crate/crate/commit/122620fc359080992ee14dca7180b2427a695b63 first

Comment (matriv) [reply .readthedocs.yml:7]: There is currently a failure:
```
File "/var/lib/jenkins/workspace/CrateDB/crate_on_pr/blackbox/docs/general/ddl/data-types.rst", line 2207, in data-types.rst
Failed example:
    cmd.process('INSERT INTO my_table (\n    title,\n    protagonist\n) VALUES (\n    \'Alice in Wonderland\',\n    {\n        "age" = \'10\',\n    }\n)')Expected:
    SQLParseException[line 8:5: no viable alternative at input 'VALUES (\n    'Alice in Wonderland',\n    {\n        "age" = '10',\n    }']
Got:
    SQLParseException[line 8:1: no viable alternative at input 'VALUES (\n'Alice in Wonderland',\n{\n"age" = '10',\n}']
```

Comment (matriv) [reply .readthedocs.yml:7]: > Looks like we need to backport [122620f](https://github.com/crate/crate/commit/122620fc359080992ee14dca7180b2427a695b63) first

I'll do it now, thx!

Comment (matriv) [reply .readthedocs.yml:7]: > There is currently a failure:
> 
> ```
> File "/var/lib/jenkins/workspace/CrateDB/crate_on_pr/blackbox/docs/general/ddl/data-types.rst", line 2207, in data-types.rst
> Failed example:
>     cmd.process('INSERT INTO my_table (\n    title,\n    protagonist\n) VALUES (\n    \'Alice in Wonderland\',\n    {\n        "age" = \'10\',\n    }\n)')Expected:
>     SQLParseException[line 8:5: no viable alternative at input 'VALUES (\n    'Alice in Wonderland',\n    {\n        "age" = '10',\n    }']
> Got:
>     SQLParseException[line 8:1: no viable alternative at input 'VALUES (\n'Alice in Wonderland',\n{\n"age" = '10',\n}']
> ```

Pushed a commit to fix this, it has been addressed for quite some time now by: https://github.com/crate/crate/pull/13886

**Comment by matriv** [comment]
@seut Could you please review this now, I've added some commits to fix issues with linkcheck, doc test and the add to project workflow.

**Review by seut — APPROVED** [review]
👍

**Pull request merged**